### PR TITLE
fix: zig 0.13.0-dev.46+3648d7df1

### PR DESCRIPTION
**Pull request by WriteNaN**
Minor change to work with zig v13

### CODE REVIEW

**Comment by giann** [comment]
Thanks.

It would be easier to review without all the reformatting. Can you try and remove those?

**Comment by WriteNaN** [comment]
sure, oops sorry for that

**Comment by WriteNaN** [comment]
sure, also tests were passing for me. i'll the debugging to see what caused the workflow to fail

**Comment by WriteNaN** [comment]
that should be good, lmk! 🙂

**Comment by WriteNaN** [comment]
that should be good, lmk! 🙂

**Comment by WriteNaN** [comment]
sorry for all this mess, I'm familiarizing myself with this codebase.

[this commit](https://github.com/buzz-language/buzz/pull/276/commits/f0d754ce2cc8a50778b371559a406b21e439804a) has most of the changes which is very minor (switching from std.ComptimeStringMap).

tests should pass now.

**Comment by giann** [comment]
No worries I'll squash all that :)

File: src/lib/buzz_os.zig [hunk 1]
```diff
@@ -133,6 +133,8 @@ fn handleSpawnError(ctx: *api.NativeCtx, err: anytype) void {
 
         error.OutOfMemory => @panic("Out of memory"),
         error.Unexpected => ctx.vm.pushError("errors.UnexpectedError", null),
+
+        else => ctx.vm.pushError("errors.UnexpectedError", null),
```
Comment (giann) [inline src/lib/buzz_os.zig:137]: You should not need to do this. If its a new error, add it to lib/error.buzz

File: src/FFI.zig [hunk 2]
```diff
@@ -12,168 +12,162 @@ const Reporter = @import("Reporter.zig");
...
-        .{ "bool", .{ .def_type = .Bool } },
+    .{ "bool", .{ .def_type = .Bool } },
 
-        .{ "void", .{ .def_type = .Void } },
-        .{ "anyopaque", .{ .def_type = .Void } },
-    },
-);
+    .{ "void", .{ .def_type = .Void } },
+    .{ "anyopaque", .{ .def_type = .Void } },
+});
```
Comment (giann) [inline src/FFI.zig:38]: The reformatting is due to the last `,` being removed. Add them back and the diff will disappear.

File: src/lib/buzz_os.zig (hunk 1 above)
Comment (WriteNaN) [reply src/lib/buzz_os.zig]: added default error for the switch handle

**Comment by WriteNaN** [comment]
that should be good?

**Comment by WriteNaN** [comment]
oh oops! haha sorry

**Comment by WriteNaN** [comment]
that should be it! :D

**Comment by giann** [comment]
The error must also exists on the buzz side. You can add it under here:

https://github.com/buzz-language/buzz/blob/main/src/lib/errors.buzz#L3

The following means we're going to push on the stack the error `errors.FileSystemError.InvalidBatchScriptArg`.
```zig
ctx.vm.pushErrorEnum("errors.FileSystemError", @errorName(err))
```

Right now zig errors have their counterpart on the buzz side but I plan on simplifying those a little on the buzz side (https://github.com/buzz-language/buzz/issues/157)

**Comment by WriteNaN** [comment]
> The error must also exists on the buzz side. You can add it under here:
> 
> https://github.com/buzz-language/buzz/blob/main/src/lib/errors.buzz#L3
> 
> The following means we're going to push on the stack the error `errors.FileSystemError.InvalidBatchScriptArg`.
> 
> ```zig
> ctx.vm.pushErrorEnum("errors.FileSystemError", @errorName(err))
> ```
> 
> Right now zig errors have their counterpart on the buzz side but I plan on simplifying those a little on the buzz side (#157)

sorry I missed that, added now. Yes that is true. but I think the access to know what the error exactly is on buzz is also great.

**Comment by giann** [comment]
Huh. I'm not reproducing the CI error on my mac. Seems like the mir building step is doing it on a different cpu from the following steps.

**Comment by giann** [comment]
Merging anyway, i'll temporarily disable the macos test if this persists

**Pull request merged**